### ISSUE TITLE
OpInfo tests should compare gpu to cpu implementations

### ISSUE THREAD

**Issue by fdrocha** [pytorch/pytorch#84538]
I believe that currently we do not have any OpInfo based tests that compare the results of gpu to cpu implementations.
Seems like this would be very useful to have, and shouldn't be too hard to implement.
You could just take the intersection of cpu and gpu dtypes for each OpInfo and compare outputs.

(FYI, I came across a discrepancy between gpu and cpu implementations (of `nn.functiona.grid_sample`) which made me think of this. )

**Comment by mruberry** [comment]
We don't have a test that directly compares CPU vs GPU results, that's true, but we do validate their consistency indirectly with tests like:

https://github.com/pytorch/pytorch/blob/fe353e1413e2262993fb71dba7317f21bc6fc3bc/test/test_ops.py#L151

which compare both CPU and GPU implementations to a NumPy reference implementation. Of course that test only works when a NumPy reference has been added to the OpInfo. 

We thought about comparing CPU vs GPU for consistency directly when the op doesn't have a NumPy reference specified, and it seems reasonable to do to cover this consistency gap, but ideally most OpInfos would have NumPy reference implementations. I imagine we could even derive NumPy references for operators now based on their primTorch decompositions...

**Comment by Lezcano** [comment]
I think it would be good to have a test to cover this gap until the PrimTorch coverage is higher.

**Comment by fdrocha** [comment]
FYI, only 37% of OpInfos currently have numpy references:

```python
import torch.testing._internal.common_methods_invocations as cmi
len([op for op in cmi.op_db if op.ref is not None])/len(cmi.op_db) # output 0.36644951140065146
```

There's an extra 15% of OpInfos that have a PrimTorch ref but with a parent without a numpy Ref
```
len([op for op in cmi.python_ref_db if op.ref is None])/len(cmi.op_db) # Output 0.1465798045602606
```

That leaves 47% of OpInfos where Cpu and CUDA implementations possibly diverge.

**Comment by mruberry** [comment]
Yes, for OpInfos without a NumPy/SciPy reference implementation we could implement this consistency check
